### PR TITLE
Improve Send-MailMessage tests

### PR DESCRIPTION
**Pull request by ThreeFive-O**
## PR Summary

Use newer Pester syntax on Send-MailMessage cmdlet tests:

- Use Pester test cases
- Add logic to store current mailbox content before running tests
- Refactoring of code

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [ ] Not Applicable or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1 [hunk 1]
```diff
@@ -1,171 +1,169 @@
...
+        $tc = New-Object -TypeName System.Net.Sockets.TcpClient -ArgumentList "localhost", 25
+        $rv = $tc.Connected
+        $tc.Close()
+        return $rv
+    }
+    catch
+    {
+        return $false
+    }
+}
```
Comment (adityapatwardhan) [inline test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1]: Please add a newline between these lines.

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

**Comment by adityapatwardhan** [comment]
@ThreeFive-O It seems that the test is marked as pending on Linux. Is that expected?

I would prefer not to have a different name when the test is pending versus when it succeeds. This help in tracking history of tests.

**Comment by ThreeFive-O** [comment]
@adityapatwardhan I can't really see the full output of the PowerShell-CI-linux test run from Pester. The test should only be marked as pending, if no SMTP server is available on the CI or if the user password can't be retrieved from /etc/password.
It's the same logic as the old test. Do you have any information if the PowerShell-CI-linux hosts a local Postfix SMTP server?

**Comment by adityapatwardhan** [comment]
@ThreeFive-O You can go to logs and see that both tests are marked as pending:

`
2018-09-15T19:08:37.4876307Z   Describing Send-MailMessage
`
`
2018-09-15T19:08:37.5576199Z ??
`

If there are not running on Linux-CI then maybe add more logging with `Write-Verbose -Verbose` for the reason on skipping.

@TravisEz13  would the following work on Linux-CI hosts?

```PowerShell
$tc = New-Object -TypeName System.Net.Sockets.TcpClient -ArgumentList "localhost", 25
```

**Comment by TravisEz13** [comment]
It would only work if a mail host has been installed.

**Comment by ThreeFive-O** [comment]
I think that explains why the test is marked as pending. The daily build of the master branch is also marked as pending:
```
2018-09-28T10:15:30.5317964Z   Describing Basic Send-MailMessage tests
2018-09-28T10:15:30.6222472Z ??
```
@TravisEz13 and @adityapatwardhan How should we proceed? Make another PR which installs Postfix mail server on the PowerShell-CI-Linux? Honestly I'm not very familiar with that.

On the other hand I'm working on a solution for #3950. But this is time-consuming, given the fact I want to enhance the test for the cmdlet first.

**Comment by anmenaga** [comment]
@TravisEz13 @adityapatwardhan please provide your feedback on questions in the last post.

**Comment by TravisEz13** [comment]
I would recommend adding a setup script for postfix and adding a step to the linux.yml to install it.  Detect if it's there and skip the test if it's not there.  This would give us the most flexibility.

**Comment by ThreeFive-O** [comment]
Superseded by PR #8859

**Pull request closed without merging**